crates/provider/src/quorum/transport.rs

```diff
@@ -0,0 +1,240 @@
+use crate::quorum::{
+    Quorum,
+    QuorumError,
+};
+use alloy_json_rpc::{
+    RequestPacket,
+    Response,
+    ResponsePacket,
+    ResponsePayload,
+    RpcError,
+};
+use alloy_transport::{
+    BoxTransport,
+    TransportError,
+    TransportErrorKind,
+};
+use futures::FutureExt;
+use std::{
+    pin::Pin,
+    task::Poll,
+};
+use tower::Service;
+
+#[derive(Clone)]
+pub struct QuorumTransport {
+    transports: Vec<WeightedTransport>,
```

Comment:
```suggestion
    transports: Arc<[WeightedTransport]>,
```

or an Rc works too

dont think we'll add more once its constructed. should make the clones cheaper below